src/numeric-types.c

```diff
@@ -112,11 +198,18 @@ DEFINE_NUMERIC_WITH_BYTESWAP (double_be, double, gint32,  GINT64,  BE)
 
 #if HAVE_LIBDFP
 extern void register_printf_dfp (void);
+#endif
 
 __attribute__ ((constructor))
 static void
 numeric_types_init (void)
 {
+    numeric_type_register_static (numeric_decimal128_get_type (),
+                                  "decimal128",
+                                  sizeof (numeric_decimal128),
+                                  NUMERIC_BYTE_ORDER_UNKNOWN);
+
+#if HAVE_LIBDFP
```

Comment:
Registration should be performed lazily in `numeric_{type}_get_type`, but that would require redefining GLib macros for boxed types registration.